crates/wasi-common/yanix/src/filetime.rs

```diff
@@ -1,34 +1,50 @@
-//! This internal module consists of helper types and functions for dealing
+//! This module consists of helper types and functions for dealing
 //! with setting the file times (mainly in `path_filestat_set_times` syscall for now).
 //!
 //! The vast majority of the code contained within and in platform-specific implementations
 //! (`super::linux::filetime` and `super::bsd::filetime`) is based on the [filetime] crate.
 //! Kudos @alexcrichton!
 //!
 //! [filetime]: https://github.com/alexcrichton/filetime
-use crate::Result;
-use std::convert::TryInto;
+use std::io::Result;
 
-pub(crate) use super::sys_impl::filetime::*;
+pub use super::sys::filetime::*;
 
 cfg_if::cfg_if! {
     if #[cfg(not(target_os = "emscripten"))] {
         fn filetime_to_timespec(ft: &filetime::FileTime) -> Result<libc::timespec> {
             Ok(
                 libc::timespec {
                     tv_sec: ft.seconds(),
-                    tv_nsec: ft.nanoseconds().try_into()?,
+                    // `as` cast is fine here since `libc::timespec::tv_nsec` is `i64`
+                    // and `ft.nanoseconds()` is `u32` so no possibility for an overflow
+                    tv_nsec: ft.nanoseconds() as _,
```

Comment:
Slightly more verbose, but avoids the need for a two-line comment: `i64::from(ft.nanoseconds())` :-).

Reply:
Excellent, thanks!